File: starters/java/credit/Credit.java
// TODO: 实现 Credit.java
import java.util.Scanner;

public class Credit {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // 获取用户输入的信用卡号
        System.out.print("Number: ");
        long cardNumber = scanner.nextLong();

        // 使用 Luhn 算法验证卡号

        // 判断卡类型：AMEX, MASTERCARD, VISA 或 INVALID

        scanner.close();
    }

    // Luhn 算法验证
    private static boolean isValidLuhn(long number) {
        // TODO: 实现 Luhn 算法
        return false;
    }

    // 获取卡号长度
    private static int getLength(long number) {
        // TODO: 实现
        return 0;
    }

    // 获取卡号前两位
    private static int getFirstTwoDigits(long number) {
        // TODO: 实现
        return 0;
    }
}

File: starters/python/credit/credit.py
# TODO: 实现 credit.py


def get_card_number():
    """获取用户输入的信用卡号"""
    while True:
        try:
            return int(input("Number: "))
        except ValueError:
            pass


def main():
    card_number = get_card_number()

    # 使用 Luhn 算法验证卡号

    # 判断卡类型：AMEX, MASTERCARD, VISA 或 INVALID
    print("INVALID")


if __name__ == "__main__":
    main()
